xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/user/impl/xwiki/MyBasicAuthenticator.java

```diff
@@ -56,6 +76,8 @@ public boolean processLogin(SecurityRequestWrapper request, HttpServletResponse
             // show the basic authentication window again.
             showLogin(request.getCurrentRequest(), response);
             return true;
+        } else {
+            getObservationManager().notify(new UserAuthenticatedEvent(getContextUserReference(principal)), null);
```

Comment:
Seems to me this should be located in checkLogin instead. In general your  criteria to find the location should be "right after the call to request.setUserPrincipal() which register the user in the session" IMO.

Reply:
Since `MyBasicAuthenticator#checkLogin` is a static method, I can't access getObservationManger nor getContextUserReference. However, `MyBasicAuthenticator#checkLogin` is only used twice and I can add this event sending right after it each time and that is what I did in the next commit.